src/promptflow/promptflow/_core/tool_meta_generator.py

```diff
@@ -170,17 +192,34 @@ def load_python_module(content, source=None):
         exec(content, m.__dict__)
         return m
     except Exception as e:
-        msg = f"Parsing python got exception: {e}"
-        raise PythonParsingError(msg) from e
+        error_type_and_message = f"({e.__class__.__name__}) {e}"
+        raise PythonParsingError(
+            message_format="Failed to load python module. Python parsing failed: {error_type_and_message}",
+            error_type_and_message=error_type_and_message,
+        ) from e
 
 
 def collect_tool_function_in_module(m):
     tools = collect_tool_functions_in_module(m)
     if len(tools) == 0:
-        raise NoToolDefined("No tool found in the python script.")
+        raise NoToolDefined(
+            message_format=(
+                "The number of tools defined is illegal. "
```

Comment:
I feel like we can remove this line. It makes the error message long and hard to read.

Reply:
Done.